src/engine/strat_engine/blockdev.rs

```diff
@@ -328,3 +328,125 @@ impl BlockDev {
         self.used.request(size)
     }
 }
+
+#[derive(Debug)]
+pub struct BlockDevMgr {
+    pub block_devs: HashMap<PathBuf, BlockDev>,
+}
+
+impl BlockDevMgr {
+    pub fn new(block_devs: Vec<BlockDev>) -> BlockDevMgr {
+        BlockDevMgr {
+            block_devs: block_devs.into_iter().map(|bd| (bd.devnode.clone(), bd)).collect(),
+        }
+    }
+
+    pub fn add(&mut self,
+               pool_uuid: &PoolUuid,
+               paths: &[&Path],
+               force: bool)
+               -> EngineResult<Vec<PathBuf>> {
+        let devices = try!(resolve_devices(paths));
+        let bds = try!(initialize(pool_uuid, devices, MIN_MDA_SECTORS, force));
+        let bdev_paths = bds.iter().map(|p| p.devnode.clone()).collect();
+        for bd in bds {
+            self.block_devs.insert(bd.devnode.clone(), bd);
+        }
+        Ok(bdev_paths)
+    }
+
+    pub fn remove_all(mut self) -> EngineResult<()> {
+        for (_, bd) in self.block_devs.drain() {
+            try!(bd.wipe_metadata());
+        }
+        Ok(())
+    }
+
+    // Unused space left on blockdevs
+    pub fn avail_space(&self) -> Sectors {
+        self.block_devs
+            .values()
+            .map(|bd| bd.available())
+            .sum()
+    }
+
+    /// Allocate segments equal to the requested sectors from the
+    /// blockdevs. All or None.
```

Comment:
The header comment is a bit cryptic. Maybe:

If available space is less than size, return None, else return the segments allocated.